src/dd/Edge.cpp

```diff
@@ -24,6 +24,13 @@ namespace dd {
 ///                      \n General purpose methods \n
 ///-----------------------------------------------------------------------------
 
+// template <class Node> bool Edge<Node>::isIdentity() const {
+//   if constexpr (std::is_same_v<Node, mNode> || std::is_same_v<Node, dNode>) {
+//     return isTerminal() && w != Complex::zero();
+//   }
+//   return false;
+// }
```

Comment:
This looks reasonable as it is. You can just uncomment it and use the respective function.
Potentially, it could be worth adding an additional flag `upToGlobalPhase` that defaults to `true`. If that flag is set to false, `w` has to be equal to `Complex::one()` for the method to return true.